### PR TITLE
Dev/android/text annotation

### PR DESCRIPTION
**Pull request by atillaturkmen**
I have implemented backend connection for text annotations. Further information can be found in the issue https://github.com/bounswe/bounswe2022group7/issues/576.

### CODE REVIEW

File: android/lib/network/annotation/post_annotation_service.dart [hunk 1]
```diff
@@ -46,8 +42,53 @@ Future<bool> postAnnotation(Map<String, dynamic> annotation) async {
...
+      ],
+      "target": {
+        "source": annotation['source'],
+        "selector": {
+          "type": "TextPositionSelector",
+          "start": annotation['start'],
+          "end": annotation['end'],
+        }
+      },
+      "id": "c_#${uuid.v4()}"
```
Comment (canatakan) [inline android/lib/network/annotation/post_annotation_service.dart:81]: Could you please update this as follows:

- For art items: `a{id}`
- For events `e{id}`
- Fo discussion posts `d{id}`

File: android/lib/network/annotation/get_annotation_service.dart [hunk 2]
```diff
@@ -34,6 +34,33 @@ Future<List<ImageAnnotation>> getAnnotationsNetwork() async {
...
+      Uri.parse(Uri.encodeFull(annotationsURL)),
+      headers: {'Content-Type': 'application/json; charset=UTF-8'},
+    );
+    if (response.statusCode != 200) {
+      return [];
+    }
+    List<dynamic> data = json.decode(response.body);
+    List<dynamic> textAnnotations = [];
+    for (var annotation in data) {
+      if (annotation['id'][0] == "c") {
```
Comment (canatakan) [inline android/lib/network/annotation/get_annotation_service.dart:51]: You should also update this if check after your update.

**Review by canatakan — CHANGES_REQUESTED** [review]
After you make the updates, you can also change the if check in `line 22` as we decided on a new text annotation standard. Thank you for your effort.

File: android/lib/network/annotation/post_annotation_service.dart (hunk 1 above)
Comment (atillaturkmen) [reply android/lib/network/annotation/post_annotation_service.dart]: I have made the requested changes in the commit https://github.com/bounswe/bounswe2022group7/pull/622/commits/a98e85288c19e389f381ea0f93a6df5a7e7d492f.

File: android/lib/network/annotation/get_annotation_service.dart (hunk 2 above)
Comment (atillaturkmen) [reply android/lib/network/annotation/get_annotation_service.dart]: I have made the requested changes in the commit https://github.com/bounswe/bounswe2022group7/commit/a98e85288c19e389f381ea0f93a6df5a7e7d492f.

**Review by canatakan — APPROVED** [review]
I used your code and the text annotations are created and posted to our annotation server. This branch provides the necessary functionality. I approve your updates.

**Pull request merged**